### PR TITLE
First batch of shortcuts to HLO ops

### PR DESCRIPTION
**Pull request by mofeing**
- [x] implement ops
- [x] test ops
- ~rewrite `TracedR*` methods~

### CODE REVIEW

File: src/Ops.jl [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+) where {T,N}
+    res = MLIR.IR.result(
+        stablehlo.complex(
+            real.mlir_data,
+            imag.mlir_data;
+            result=mlir_type(TracedRArray{T,N}, size(real)),
+            location,
+        ),
+    )
+    return TracedRArray{T,N}((), res, size(real))
```
Comment (Pangoraw) [inline src/Ops.jl:411]: I think this is missing

```suggestion
    return TracedRArray{Complex{T},N}((), res, size(real))
```

File: src/Ops.jl [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+) where {T}
+    res = MLIR.IR.result(
+        stablehlo.complex(
+            real.mlir_data,
+            imag.mlir_data;
+            result=mlir_type(TracedRArray{T,0}, ()),
+            location,
+        ),
+    )
+    return TracedRNumber{T}((), res)
```
Comment (Pangoraw) [inline src/Ops.jl:429]: Same as above

```suggestion
    return TracedRNumber{Complex{T}}((), res)
```

**Review by Pangoraw — COMMENTED** [review]
nice!

**Comment by mofeing** [comment]
# Update

I'm done with this first batch of ops. Now I will write some tests and it should be done.

**Comment by mofeing** [comment]
> so CI apparently fails

should be fixed now

I've left some ops without testing due to bugs (mainly #196 but also some weird problem generating the IRCode for `Ops.polygamma`) or complex test configuration (like `send`, `recv`, ...)

**Comment by wsmoses** [comment]
test_broken is fine if it doesn't crash, a commented or disabled test is also fine

**Comment by mofeing** [comment]
then I will leave it commented because some crash

**Comment by mofeing** [comment]
@wsmoses any idea of how to solve macOS-x86?

**Review by wsmoses — APPROVED** [review]

File: src/Ops.jl [hunk 3]
```diff
@@ -0,0 +1,923 @@
...
+
+struct Token
+    mlir_data::MLIR.IR.Value
+end
+
+# constant ops
+function constant(
+    x::DenseArray{T,N};
+    location=MLIR.IR.Location(
+        "stablehlo.constant", MLIR.IR.Location(@__FILE__, @__LINE__, 0)
```
Comment (wsmoses) [inline src/Ops.jl:18]: we still should refactor this into a location macro

```
macro location()
return MLIR.IR.Location(FILE, LINE)
end
```

which is then used everywhere (and can be overridden to be that more full back trace as mentioned

Comment (mofeing) [reply src/Ops.jl:18]: yes, I have sth in mind but will open another PR

**Pull request merged**